# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/nsHttpConnection.cpp
# Commit: 8e7c97a37073
# Full Hash: 8e7c97a37073a14ba832e2f5c27f2c11b7787ef5
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2023-03-15 16:26:49
# Regressor Bug: 1805372
# File Overlap Count: 1
# Description:
#   Bug 1805372 - Pass effectiveTRRMode to nsHttpChannel from socketTransport/DnsAndConnectSocket r=necko-reviewers,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D164539
# ==============================================================================

diff -r 5abfd5ad2e35 -r 8e7c97a37073 netwerk/protocol/http/nsHttpConnection.cpp
--- a/netwerk/protocol/http/nsHttpConnection.cpp	Wed Mar 15 12:41:26 2023 +0000
+++ b/netwerk/protocol/http/nsHttpConnection.cpp	Wed Mar 15 13:00:46 2023 +0000
@@ -2326,6 +2326,22 @@
   return val;
 }
 
+nsIRequest::TRRMode nsHttpConnection::EffectiveTRRMode() {
+  nsIRequest::TRRMode mode = nsIRequest::TRR_DEFAULT_MODE;
+  if (mSocketTransport) {
+    mSocketTransport->GetEffectiveTRRMode(&mode);
+  }
+  return mode;
+}
+
+TRRSkippedReason nsHttpConnection::TRRSkipReason() {
+  TRRSkippedReason reason = nsITRRSkipReason::TRR_UNSET;
+  if (mSocketTransport) {
+    mSocketTransport->GetTrrSkipReason(&reason);
+  }
+  return reason;
+}
+
 bool nsHttpConnection::GetEchConfigUsed() {
   bool val = false;
   if (mSocketTransport) {